pkg/sbom/spdx/marshal.go

```diff
@@ -416,6 +416,16 @@ func (m *Marshaler) normalizeLicenses(licenses []string) (string, []*spdx.OtherL
 	var otherLicenses = make(map[string]*spdx.OtherLicense) // licenseID -> OtherLicense
 
 	license := strings.Join(lo.Map(licenses, func(license string, index int) string {
+		// We need to save text licenses before normalization,
+		// because it is impossible to handle all cases possible in the text.
+		// as an example, parse a license with 2 consecutive tokens (see https://github.com/aquasecurity/trivy/issues/8465)
+		if strings.HasPrefix(license, licensing.LicenseTextPrefix) {
+			license = strings.TrimPrefix(license, licensing.LicenseTextPrefix)
+			otherLicense := m.newOtherLicense(license, true)
+			otherLicenses[otherLicense.LicenseIdentifier] = otherLicense
+			license = otherLicense.LicenseIdentifier
```

Comment:
I don't think we need subsequent processes. `()` is needed for compound licenses. What do you think?

```suggestion
			return otherLicense.LicenseIdentifier
```